web/packages/teleport/src/Roles/RoleEditor/StandardEditor.tsx

```diff
@@ -419,12 +437,166 @@ export function ServerAccessSpecSection({
   );
 }
 
-function KubernetesAccessSpecSection() {
-  // TODO(bl-nero): add the Kubernetes section
-  return null;
+export function KubernetesAccessSpecSection({
+  value,
+  isProcessing,
+  onChange,
+}: SectionProps<KubernetesAccessSpec>) {
+  return (
+    <>
+      <FieldSelectCreatable
+        isMulti
+        label="Groups"
+        isDisabled={isProcessing}
+        formatCreateLabel={label => `Group: ${label}`}
+        components={{
+          DropdownIndicator: null,
+        }}
+        value={value.groups}
+        onChange={groups => onChange?.({ ...value, groups })}
+      />
+
+      <Text typography="body3" mb={1}>
+        Labels
+      </Text>
+      <LabelsInput
+        disableBtns={isProcessing}
+        labels={value.labels}
+        setLabels={labels => onChange?.({ ...value, labels })}
+      />
+
+      <Flex flexDirection="column" gap={3} mt={3}>
+        {value.resources.map((resource, iRes) => (
+          <KubernetesResourceView
+            key={iRes}
```

Comment:
[Using indexes as keys is not recommended](https://react.dev/learn/rendering-lists#why-does-react-need-keys), this is particularly true for this form as you can remove an item at any position from the model.

Would it be possible to add a `uuid` field to `KubernetesResourceModel`, generate it in `newKubernetesResourceModel` and then make sure it doesn't make its way to the backend? I haven't looked at the code too closely yet, but I assume at some point we need to map values from the UI to some kind of yaml.

Reply:
I suppose the uuid field could be just ignored within `roleEditorModelToRole`?

Reply:
Oh, crap. I'm well aware that this is an anti-pattern (as a matter of fact, I was commenting on the similar thing in another PR), but somehow still fell into this trap here. :) Yes, I'll add an ID to the editor model.